model_analyzer/config/generate/perf_analyzer_config_generator.py

```diff
@@ -377,8 +377,24 @@ def _extract_text_input_length(
     def _update_perf_config_based_on_parameter_combination(
         self, perf_config: PerfAnalyzerConfig, parameter_combination: Dict
     ) -> None:
+        if "request-parameter" in parameter_combination:
+            request_parameter = parameter_combination["request-parameter"]
+            max_token = self._extract_max_token_from_request_parameter(
+                request_parameter
+            )
+            parameter_combination["request-period"] = (
+                max_token if max_token < 10 else 10
```

Comment:
Is there reason why we are capping request period to 10?

Reply:
Good catch. This is stale code before I added the RCS option for `request-period`. I've updated the code to use `request-period`